x/shared/types/params.go

```diff
@@ -206,3 +247,34 @@ func validateGracePeriodOffsetBlocksIsLessThanNumBlocksPerSession(params *Params
 	}
 	return nil
 }
+
+// validateSupplierUnbondingPeriodIsGreaterThanCumulatedProofWindowCloseBlocks
+// validates that the SupplierUnbondingPeriod blocks is greater than the cumulated
+// proof window close blocks.
+// It ensures that a supplier cannot unbond before the pending claims are settled.
+func validateSupplierUnbondingPeriodIsGreaterThanCumulatedProofWindowCloseBlocks(params *Params) error {
+	cumulatedProofWindowCloseBlocks := GetCumulatedProofWindowCloseBlocks(params)
+	supplierUnbondingPeriodBlocks := params.SupplierUnbondingPeriod * params.NumBlocksPerSession
+
+	if supplierUnbondingPeriodBlocks < cumulatedProofWindowCloseBlocks {
+		return ErrSharedParamInvalid.Wrapf(
+			"SupplierUnbondingPeriod (%v session) (%v blocks) must be greater than the cumulated ProofWindowCloseOffsetBlocks (%v)",
+			params.SupplierUnbondingPeriod,
+			supplierUnbondingPeriodBlocks,
+			cumulatedProofWindowCloseBlocks,
+		)
+	}
+
+	return nil
+}
+
+// GetCumulatedProofWindowCloseBlocks returns the total number of blocks required
+// to pass the proof window close of a session.
+// Using shared.GetProofWindowCloseOffsetHeight is not possible because of the
+// circular dependency between the shared and session modules.
```

Comment:
```suggestion
// GetSessionStartToProofWindowCloseBlocks returns the total number of blocks 
// from the moment a session starts until the proof window closes.
// NB: Using shared.GetProofWindowCloseOffsetHeight is not possible because of the
// circular dependency between the shared and session modules.
```

Reply:
s/SessionStart/SessionEnd/